src/Couchbase.Lite.Shared/Serialization/FleeceMutableArray.cs

```diff
@@ -135,26 +149,37 @@ public void Set(int index, object val)
                 throw new ArgumentNullException(nameof(val));
             }
 
+            if (index < Count) {
+                PopulateArr();
+            }
+
             Mutate();
+            using (var encoded = val.FLEncode()) { 
+                //Convert object into FLValue
+                var flValue = NativeRaw.FLValue_FromData((FLSlice)encoded, FLTrust.Trusted);
+                Native.FLSlot_SetValue(Native.FLMutableArray_Set(ar, (uint)index), flValue);
+            }
```

Comment:
This will probably need to be reworked later.  There is a whole suite of `FLSlot_Set*` functions that are available and can be used by writing a method similar to `FLEncode()` that calls the appropriate `FLSlot_Set*` methods.  This avoids encoding upfront.